File: src/main/java/br/com/sd/streams/BoardInputStream.java
package br.com.sd.streams;

import br.com.sd.entitys.Board;
import br.com.sd.services.BoardService;

import java.io.IOException;
import java.io.InputStream;

public class BoardInputStream extends InputStream {

    private InputStream in;

    public BoardInputStream(InputStream in) {
        this.in = in;
    }

    @Override
    public int read() throws IOException {
        return in.read();
    }

    public Board readBoard() throws IOException {
        int width = readInt();
        int height = readInt();

        int[][] grid = new int[width][height];

        for (int y = 0; y < width; y++) {
            for (int x = 0; x < height; x++) {
                grid[y][x] = readInt();
            }
        }

        return new Board(grid);
    }

    private int readInt() throws IOException {
        int b1 = in.read();
        int b2 = in.read();
        int b3 = in.read();
        int b4 = in.read();

        if ((b1 | b2 | b3 | b4) < 0) {
            throw new IOException("Fim do stream inesperado");
        }

        return ((b1 << 24) |
                (b2 << 16) |
                (b3 << 8) |
                b4);
    }
}

File: pyapp/streams/boardInputStream.py
from entitys.pixel import Pixel
from entitys.board import Board


class BoardInputStream:
    def __init__(self, inputStream):
        self.inputStream = inputStream

    def read(self):
        return self.inputStream.read()

    def readBoard(self):
        width = self.readInt()
        height = self.readInt()

        grid = []

        for y in range(height):
            grid.append([])
            for _ in range(width):
                grid[y].append(self.readInt())

        return Board(grid, width, height)

    def readInt(self):
        integer_b = self.inputStream.read(4)
        integer = int.from_bytes(integer_b, byteorder='big')
        return integer

